### PR TITLE
WIP Add Alert Module

### PR DESCRIPTION
**Pull request by t-wihatz**
Implements alert module for the UWP platform.

Calling the alert is the same as Facebook's documentation, the user calls Alert.alert and passes in strings for title and message, as well as array for the buttons. The number of buttons is variable, with up to 3 buttons allowed to be specified. The last button is mapped to Primary, the second-to-last to Close, and the third-to-last button to Secondary.

Additionally added an Alert Example test page for RNTester, which shows examples of different Alerts being called.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2935)

### CODE REVIEW

File: vnext/ReactUWP/Base/UwpReactInstance.cpp [hunk 1]
```diff
@@ -218,6 +219,11 @@ std::vector<facebook::react::NativeModuleDescription> GetModules(
         return std::make_unique<react::windows::AppThemeModule>(
             std::move(appTheme));
       },
+
+      messageQueue);
```
Comment (kmelmon) [inline vnext/ReactUWP/Base/UwpReactInstance.cpp]: >messageQueue [](start = 6, length = 12)

This should move one line up

File: vnext/src/index.ts [hunk 2]
```diff
@@ -16,3 +16,4 @@ export * from './Libraries/Components/View/ViewWindowsProps';
 export * from './Libraries/Components/View/ViewWindows';
 export * from './Libraries/AppTheme/AppTheme';
 export * from './Libraries/AppTheme/AppThemeTypes';
+export * from './Libraries/Modules/Alert/Alert.uwp';
```
Comment (kmelmon) [inline vnext/src/index.ts]: You will also need to add a line like this to index.windows.ts

File: vnext/src/Libraries/Modules/Alert/Alert.uwp.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * @format
+ */
+'use strict';
```
Comment (kmelmon) [inline vnext/src/Libraries/Modules/Alert/Alert.uwp.ts]: This file will need to be renamed to Alert.windows.ts

File: vnext/src/RNTester/RNTesterList.windows.ts [hunk 4]
```diff
@@ -139,6 +139,10 @@ const APIExamples: Array<IRNTesterExample> = [
     key: 'AppStateExample',
     module: require('react-native/RNTester/js/AppStateExample'),
   },
+  {
+    key: 'AlertExampleUwp',
+    module: require('./AlertExampleUwp'),
```
Comment (acoates-ms) [inline vnext/src/RNTester/RNTesterList.windows.ts]: Still dont think we should write a new one of these.  We should use the alert tester page thats already part of react-native/RNTester

Comment (t-wihatz) [reply vnext/src/RNTester/RNTesterList.windows.ts]: In order to do that alert.windows.ts would have to be moved into facebook's alert.js, which I plan on doing in a separate pull request. This tester page is only temporary until that happens

File: vnext/src/Libraries/Modules/Alert/Alert.windows.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * @format
+ */
+'use strict';
+
+import {NativeModules} from 'react-native';
```
Comment (kmelmon) [inline vnext/src/Libraries/Modules/Alert/Alert.windows.ts]: Please file an issue to move this code into react-native, and add a TODO here tracking it

**Pull request merged**